test/e2e_node/cpu_manager_test.go

```diff
@@ -596,17 +597,30 @@ func runCfsQuotaGuPods(ctx context.Context, f *framework.Framework, disabledCPUQ
 	var err error
 	var ctnAttrs []ctnAttribute
 	var pod1, pod2, pod3 *v1.Pod
-	var cleanupPods []*v1.Pod
+	podsToClean := make(map[apitypes.UID]*v1.Pod) // pod.UID -> pod
+
+	cleanupTestPod := func(ctx2 context.Context, pod *v1.Pod) {
+		framework.Logf("deleting pod: %s/%s", pod.Namespace, pod.Name)
+		deletePodSyncByName(ctx2, f, pod.Name)
+		waitForContainerRemoval(ctx2, pod.Spec.Containers[0].Name, pod.Name, pod.Namespace)
+		framework.Logf("deleted pod: %s/%s", pod.Namespace, pod.Name)
+	}
+
+	deleteTestPod := func(pod *v1.Pod) {
+		// waitForContainerRemoval takes "long" to complete; if we use the parent ctx we get a
+		// 'deadline expired' message and the cleanup aborts, which we don't want.
+		cleanupTestPod(context.TODO(), pod)
```

Comment:
What if there are failures in cleanup? With `context.TODO()` we could end up in indefinite waits in case of failures so perhaps it is better to use ` context .WithTImeout `with some reasonable timeout?

Reply:
good point, waiting for hours just to see the lane fail is bad dev UX anyway. Adding a timeout.